src/client/datascience/jupyter/kernels/kernelCommandListener.ts

```diff
@@ -194,8 +194,9 @@ export class KernelCommandListener implements IDataScienceCommandListener {
             if (exc instanceof JupyterKernelPromiseFailedError && kernel) {
                 // Old approach (INotebook is not exposed in IKernel, and INotebook will eventually go away).
                 const notebook = await this.notebookProvider.getOrCreateNotebook({
-                    resource: notebookDocument.uri,
-                    identity: notebookDocument.uri,
+                    resource: kernel.notebookDocument.uri,
+                    // TODO: This would be wrong for interactive window;
```

Comment:
@joyceerhl /cc
I'll create an issue for this.
I think we're passing the wrong `resource` at this point for interactive window.